// @hilma/secrets-handler/getConnectorSecret.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @param {*} secretObj from type connector
 * @returns secret data from env file
 */
const getConnectorSecret = (secretObj) => __awaiter(void 0, void 0, void 0, function* () {
    return {
        username: typeof secretObj.username == "string" ? secretObj.username : process.env[`${secretObj.alias}_USER`],
        password: typeof secretObj.password == "string" ? secretObj.password : process.env[`${secretObj.alias}_PASSWORD`],
        engine: typeof secretObj.engine == "string" ? secretObj.engine : process.env[`${secretObj.alias}_ENGINE`],
        host: typeof secretObj.host == "string" ? secretObj.host : process.env[`${secretObj.alias}_HOST`],
        port: typeof secretObj.port == "number" ? secretObj.port : Number(process.env[`${secretObj.alias}_PORT`]),
        dbname: typeof secretObj.dbname == "string" ? secretObj.dbname : process.env[`${secretObj.alias}_NAME`],
    };
});
exports.default = getConnectorSecret;
